src/database/database.cpp

```diff
@@ -30,33 +30,20 @@ bool Database::connect() {
 	bool reconnect = true;
 	mysql_options(handle, MYSQL_OPT_RECONNECT, &reconnect);
 
-	// connects to database
```

Comment:
I believe you can actually simplify this and keep the separation of functions and simplify them like this:

```cpp
		bool Database::connect() {
			return connect(
				handle,
				g_configManager().getString(MYSQL_HOST),
				g_configManager().getString(MYSQL_USER),
				g_configManager().getString(MYSQL_PASS),
				g_configManager().getString(MYSQL_DB),
				g_configManager().getNumber(SQL_PORT),
				g_configManager().getString(MYSQL_SOCK),
				0
			);
		}

		bool Database::connect(
			const std::string* host,
			const std::string* user,
			const std::string* password,
			const std::string* database,
			uint32_t port,
			const std::string* sock
		) {
			// connection handle initialization
			handle = mysql_init(nullptr);
			if (!handle) {
				SPDLOG_ERROR("Failed to initialize MySQL connection handle.");
				return false;
			}

			if (host->empty() || user->empty() || password->empty() || database->empty() || port <= 0) {
				SPDLOG_WARN("MySQL host, user, password, database or port not provided");
				return false;
			}

			// automatic reconnect
			bool reconnect = true;
			mysql_options(handle, MYSQL_OPT_RECONNECT, &reconnect);

			// connects to database
			if (!mysql_real_connect(handle, host->c_str(), user->c_str(), password->c_str(), database->c_str(), port, sock->c_str(), 0)) {
				SPDLOG_ERROR("MySQL Error Message: {}", mysql_error(handle));
				return false;
			}

			DBResult_ptr result = storeQuery("SHOW VARIABLES LIKE 'max_allowed_packet'");
			if (result) {
				maxPacketSize = result->getNumber<uint64_t>("Value");
			}
			return true;
		}
```